07_distributed_systems/01_intro/server/server.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"encoding/gob"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"kv_store/protocol"
+	"log"
+	"net/http"
+	"os"
+	"runtime"
+	"sync"
+)
+
+const (
+	SERVER           = "127.0.0.1"
+	PORT             = "8000"
+	URL              = SERVER + ":" + PORT
+	STORAGE_FILEPATH = "storage"
+)
+
+// abozhenko for oz: Ok, now we have this map that fits into memory
+// Should we assume that whole dataset does not fit in memory, and
+// even on disk on a single server?
+type inMemoryStorage map[string]string
+
+var keyValueMap = make(inMemoryStorage)
+
+var storageFD *os.File
+var mutex sync.RWMutex
+
+func getHandler(w http.ResponseWriter, r *http.Request) {
+	log.Printf("getHandler. Goroutines:%v", runtime.NumGoroutine())
+	reqData := protocol.GetRequest{}
+	body, err := ioutil.ReadAll(r.Body)
+	// abozhenko for oz: Is there a best practice to handle errors
```

Comment:
@alexbozhenko not really... this degree of duplication/explicitness is encouraged